### PR TITLE
Adding Interactive Map with cases

### PR DESCRIPTION
**Pull request by ellisonleao**
Ref #5 

Questions:

- Are the lat/lng numbers going to be added on the base spreadsheet or should we get them inside the `serialize_cases` function?
- Are we going to use custom marker icons ?
- What information should the marker popups have?

I've opened the pull request with some already things changed, which includes:

- creating new `/map` route
- added a link to the map route on top of the cases list
- created initial map (no markers yet)

### CODE REVIEW

**Comment by ellisonleao** [comment]
@cuducos the changes you've asked are already made, i am just waiting for the new spreadsheet template so i can update the data model and add the markers on frontend.

**Comment by turicas** [comment]
@ellisonleao you can just add a column called "latlon" to "Casos" spreadsheet and collect data from there (it's going to be a `list` of `floats` on the model).

**Comment by ellisonleao** [comment]
@turicas are we using `|` as separator for the float numbers, just like we have for tags? e.g:`-12.12|5.34`

**Comment by turicas** [comment]
@ellisonleao I think the best would be "," because:
1- Numbers will not have "," inside them (but tags could have) since the decimal separator is "."; and
2- A lot of geolocation systems use "," so we can easily copy and paste.

**Comment by ellisonleao** [comment]
@cuducos @turicas i've added the markers code and the serialization of the latlon field. What extra information do you think we need on the maps page? Right now there is only the map and the button for the cases form.

**Comment by cuducos** [comment]
> Maybe a property function in the `Cases` data model could take care of getting that data correct

I think this sounds like a good way of implementing it. 

I tried to run your branch but it throws a `NameError: name 'cached' is not defined` in `victims/__init__.py` and there are some tests failing. So I'm afraid I can test anything further ATM. But let me know if you need anything else.

File: victims/data.py [hunk 1]
```diff
@@ -43,10 +43,17 @@ def serialize_cases(self, buffer):
...
+
+            # serializing tags
             data["tags"] = data.get("tags") or ""  # avoids getting None
             data["tags"] = data["tags"].split("|")
             data["tags"] = [tag.strip().lower() for tag in data["tags"] if tag]
 
+            # serializing  latitude and longitude
+            data["latlng"] = data["latlng"].strip() or ""
+            data["latlng"] = data["latlng"].split(",")
+            data["latlng"] = list(map(float, data["latlng"]))
```
Comment (cuducos) [inline victims/data.py]: I think this is slightly more readable, what do you think?

```suggestion
            data["latlng"] = [float(coord) for coord in data["latlng"]]
```

Comment (cuducos) [reply victims/data.py]: BTW if this column is empty this line breaks the app… we might need something for this edge case ; )

Comment (ellisonleao) [reply victims/data.py]: Yes! I've started by adding some try/excptions for `ValueError` exceptions but i am not sure what you guys prefer.

**Comment by ellisonleao** [comment]
> I think this sounds like a good way of implementing it.
actually i am not sure right now if adding a function over there will be the best.  Is there a unique element that we can use as the aggregator for same city cases (latlng, city, etc) ?

**Comment by baralhao** [comment]
hi guys

I made a dashboard with map and some informations. I will appreciate your feedback.

cya
eduardo medeiros

http://bit.ly/VitimasIntolerancia

**Pull request closed without merging**